scripts/codemods/kwarg_permissions.py

```diff
@@ -0,0 +1,158 @@
+import itertools
+from typing import Optional
+
+import libcst as cst
+import libcst.codemod.visitors as codevisitors
+import libcst.matchers as m
+from libcst import codemod
+
+from disnake import Permissions
+
+ALL_PERMISSIONS = sorted(Permissions.VALID_FLAGS.keys())
+
+PERMISSION_MATCHERS = m.OneOf(*map(m.Name, ALL_PERMISSIONS))
+
+
+def get_perm_kwargs(annotation: cst.Annotation):
+    return [
+        cst.Param(
+            cst.Name(perm),
+            annotation,
+            default=cst.Ellipsis(),
+        )
+        for perm in ALL_PERMISSIONS
+    ]
+
+
+def remove_existing_permissions(params: cst.Parameters, *, is_overload: bool) -> cst.Parameters:
+    """Remove all of the existing permissions from the kwargs of the provided cst.Parameters.
+
+    Additionally removes **kwargs if this is an overload.
```

Comment:
As far as I can tell, this isn't entirely accurate - `**kwargs` is removed further down outside of this method

Reply:
You would be right: https://github.com/DisnakeDev/disnake/pull/708/commits/8ae74da50fb8aff228661fbba1056049f9c318f8

Reply:
https://github.com/DisnakeDev/disnake/pull/708/commits/8ae74da50fb8aff228661fbba1056049f9c318f8